### DISCUSSION TITLE
Draw Border Around Image

### DISCUSSION THREAD

**Discussion by bowespublishing** [ImageMagick/ImageMagick#3759] [Help]
Is it possible to use ImageMagick to draw around an image like this?

https://ibb.co/YdVCyqT

Thank you

**Comment by fmw42** [comment]
Yes. For what platform? If Unix-based, then see my script contour at my web site http://www.fmwconcepts.com/imagemagick/index.php

**Reply by bowespublishing** [reply]
Looks great thanks!. Possible to use on Windows?

**Reply by fmw42** [reply]
For my script, only if Windows 10 Unix or Windows w/Cygwin installed.

**Reply by geekjeek** [reply]
> Yes. For what platform? If Unix-based, then see my script contour at my web site http://www.fmwconcepts.com/imagemagick/index.php

is it possible to get output as a transparent background from this script? (http://www.fmwconcepts.com/imagemagick/contour/index.php)
i test your contour script but it seems that it can not process output  of transparent background like this image:
http://www.fmwconcepts.com/misc_tests/transparent_problem4/result1.png

**Reply by fmw42** [reply]
Is result1.png an output from the script or the input? Please post both input and output so I can test that. Also what are you expecting?

**Reply by geekjeek** [reply]
The input image have transparent background: https://i.imgur.com/OZWGKE7.png
The output image with your contour script have a white background: https://i.imgur.com/3EHtOkl.png
I wanna transparent background around the contour, like this sample: http://www.fmwconcepts.com/misc_tests/transparent_problem4/result1.png

**Comment by fmw42** [comment]
Perhaps this is what you want to do. This is for IM 6 in Unix syntax

Input:

![OZWGKE7](https://user-images.githubusercontent.com/13458218/148653987-3b8022db-ae40-4ec1-bf1f-c78593a8c373.png)


```
convert OZWGKE7.png \
-bordercolor none -border 20 \
\( -clone 0 -fill white -colorize 100 \) \
\( -clone 0 -alpha extract -write mpr:alpha -morphology edgeout disk:15 \) \
-alpha off -compose over -composite \
\( mpr:alpha -morphology dilate disk:15 \) \
-alpha off -compose copy_opacity -composite \
x.png

```

![x](https://user-images.githubusercontent.com/13458218/148653984-0d920e07-4698-4c84-ba27-afeedc1c496b.png)



For IM 7, change "convert" to "magick"

You will need to download to see the transparency.

**Reply by geekjeek** [reply]
thank you for your help

**Reply by geekjeek** [reply]
> Perhaps this is what you want to do. This is for IM 6 in Unix syntax
> 
> Input:
> 
> ![OZWGKE7](https://user-images.githubusercontent.com/13458218/148653987-3b8022db-ae40-4ec1-bf1f-c78593a8c373.png)
> 
> ```
> convert OZWGKE7.png \
> -bordercolor none -border 20 \
> \( -clone 0 -fill white -colorize 100 \) \
> \( -clone 0 -alpha extract -write mpr:alpha -morphology edgeout disk:15 \) \
> -alpha off -compose over -composite \
> \( mpr:alpha -morphology dilate disk:15 \) \
> -alpha off -compose copy_opacity -composite \
> x.png
> ```
> 
> ![x](https://user-images.githubusercontent.com/13458218/148653984-0d920e07-4698-4c84-ba27-afeedc1c496b.png)
> 
> For IM 7, change "convert" to "magick"
> 
> You will need to download to see the transparency.

Is there anyway to use .jpg file (white background) as input file and output will stay like this (transparent background) ?

**Comment by fmw42** [comment]
Yes

Input test.jpg with white background:

![test_image](https://user-images.githubusercontent.com/13458218/149575415-5a33c87f-aec4-4e90-83ce-7a5e3d119c5c.jpg)

```
convert test_image.jpg -fuzz 15% -transparent white \
-bordercolor none -border 20 \
\( -clone 0 -fill white -colorize 100 \) \
\( -clone 0 -alpha extract -write mpr:alpha -morphology edgeout disk:15 \) \
-alpha off -compose over -composite \
\( mpr:alpha -morphology dilate disk:15 \) \
-alpha off -compose copy_opacity -composite \
y.png

```

![y](https://user-images.githubusercontent.com/13458218/149575456-5e2357d5-9fa7-4784-9ab2-cb3ef72c1194.png)
